### DISCUSSION TITLE
Is it possible to create a dynamic index with algorithm?

### DISCUSSION THREAD

**Discussion by swdmike** [developmentseed/titiler#626] [Q&A]
Firstly, I would like to express my gratitude to all developers. I have personally experienced significant benefits from using Titiler, and it has been incredibly inspiring.

We are considering implementing a raster calculator-type project. We would like to pass the formula from the front end to the Titler, rather than simply selecting some pre-defined indices. 

Currently, all I am thinking of is dynamically changing the formula in the index.py file.
arr = numpy.where(img.mask, (b2 - b1) / (b2 + b1), 0)
     
Is it possible?

**Comment by vincentsarago** [comment]
I think what you want to use is the "expression" query parameter 

You can check https://cogeotiff.github.io/rio-tiler/examples/Using-rio-tiler/#using-expression
